src/Functions/byteSize.cpp

```diff
@@ -0,0 +1,281 @@
+#include <Columns/ColumnVector.h>
+#include <Common/UInt128.h>
+#include <Core/Types.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <DataTypes/DataTypeFixedString.h>
+#include <DataTypes/DataTypeInterval.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+
+#include <common/logger_useful.h>
+#include <ext/range.h>
+#include <Poco/Logger.h>
+
+namespace DB
+{
+
+namespace
+{
+
+template <typename T> struct ByteSizeForNative { static constexpr const UInt64 value = sizeof(typename NativeType<T>::Type); };
+
+/** byteSize() - get the columns size in number of bytes.
+  */
+class FunctionByteSize : public IFunction
+{
+public:
+    static constexpr auto name = "byteSize";
+    static FunctionPtr create(const Context &)
+    {
+        return std::make_shared<FunctionByteSize>();
+    }
+
+    String getName() const override { return name; }
+    bool isDeterministic() const override { return false; }
+    bool isDeterministicInScopeOfQuery() const override { return false; }
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForLowCardinalityColumns() const override { return false; }
+    bool isVariadic() const override { return true; }
+    size_t getNumberOfArguments() const override { return 0; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeUInt64>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        auto result_col = ColumnUInt64::create(input_rows_count, 0);
+        auto & vec_res = result_col->getData();
+        for (const auto & arg : arguments)
+        {
+            const IColumn * column = arg.column.get();
+            const IDataType * data_type = arg.type.get();
+            byteSizeOne(data_type, column, vec_res);
+        }
+        return result_col;
+    }
+
+private:
+    static void byteSizeOne(const IDataType * data_type, const IColumn * column, ColumnUInt64::Container & vec_res)
+    {
+        size_t vec_size = vec_res.size();
+
+        UInt64 byte_size = 0;
+        if (byteSizeByDataType(data_type, byte_size))
+        {
+            for (size_t i = 0; i < vec_size; ++i)
+                vec_res[i] += byte_size;
+        }
+        else if (byteSizeByColumn(data_type, column, vec_res))
+            ;
+        else
+            LOG_WARNING(&Poco::Logger::get("FunctionByteSize"),
+                        "byteSize for \"{}\" is not supported.", data_type->getName());
+    }
+
+    static bool byteSizeByDataType(const IDataType * data_type, UInt64 & byte_size)
```

Comment:
There is
`IDataType::isValueUnambiguouslyRepresentedInFixedSizeContiguousMemoryRegion`
and
`IDataType::getSizeOfValueInMemory`